ui/component/or-asset-tree/src/index.ts

```diff
@@ -538,6 +568,99 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
         this._onNodeClicked(null, null);
     }
 
+    /**
+     * On Cancel button click, we clear everything and reset the asset tree to original content
+     *
+     * @param event
+     * @protected
+     */
+    protected _onCancelFilterChange() {
+        this._filterValue = '';
+        this._foundAssets = [];
+
+        this._loadAssets();
+    }
+
+    protected _onFilterValueChange(e: OrInputChangedEvent) {
+        this._filterValue = e.detail.value;
+        const oldFilterValue: string = e.detail.previousValue;
+
+        if (!this._filterValue.startsWith(oldFilterValue) || this._filterValue.length === 0) {
+            this._onCancelFilterChange();
+        }
+
+        if (this._searchInputTimer) {
+            clearTimeout(this._searchInputTimer);
+        }
+
+        this._searchInputTimer = setTimeout(async () => {
+            if (this._nodes && this._filterValue.length > 2) {
+                let newNodes: UiAssetTreeNode[] = this._nodes;
+                this._nodes = [];
+                this._isFiltering = true;
```

Comment:
As there doesn't seem to be a visual difference between loading and filtering the value of `_isFiltering` seems very limited. Previously when the component was loading `_nodes` was undefined so no need for additional boolean flag.

Reply:
Also shouldn't the UI state change as soon as filter input occurs? Not 1.5s later?

Reply:
I put this in place as the request was to have a visual effect of the activity of filtering. 1.5s was discussed with Don but maybe we can decrease the value a bit? @DonWillems any comments?